cpp/ql/src/Security/CWE/CWE-190/ArithmeticWithExtremeValues.ql

```diff
@@ -44,16 +44,16 @@ predicate isMinValue(Expr mie) {
   )
 }
 
-class SecurityOptionsArith extends SecurityOptions {
-  override predicate isUserInput(Expr expr, string cause) {
+class DefaultTaintTrackingOptionsArith extends DefaultTaintTrackingOptions {
```

Comment:
The configurability is bothering me a little.  The old way was one monolithic `SecurityOptions`, yet this is outside of that.

If this is only being used internally and temporarily, its probably OK.

Reply:
I don't see how this isn't the same? On `main` this query extends `SecurityOptions`, which extends `isUserInput`, which extends the sources in `DefaultTaintTracking`.

On this PR, this query extends `DefaultTaintTrackingOptionsArith`, which extends the sources in `DefaultTaintTracking`.

Isn't that the same effect?

Reply:
`DefaultTaintTrackingOptions` is in an "internal" file, so in principle users don't use it - whereas `SecurityOptions` is intentionally public (and at one point we would have encouraged users to use it).

Reply:
Mmm, yeah, this might be more problematic than I thought. This intended to be for internal use only, but if a user of the publicly exposed and deprecated default taint tracking library is not willing to rewrite their queries at this point, they'll end up using this directly, which is definitely not what we want. I think we have to wait with this until we have actually removed `DefaultTaintTracking.qll`, because at that point changes will not be exposed through a file not marked internal.